lib/src/main/scala/nl.knaw.dans.easy.stage/EasyStageDataset.scala

```diff
@@ -97,14 +98,39 @@ object EasyStageDataset extends DebugEnhancedLogging {
       foxmlContent = getDatasetFOXML(s.ownerId, emdContent)
       additionalLicenseFilenameAndMimeType <- AdditionalLicense.createOptionally(sdoDir)
       audiences <- readAudiences()
-      jsonCfgContent <- JSON.createDatasetCfg(additionalLicenseFilenameAndMimeType, audiences)
+      agreementsXmlExists = new File(s.bagitDir, "metadata/depositor-info/agreements.xml").exists()
+      messageFromDepositorExists = new File(s.bagitDir, "metadata/depositor-info/message-from-depositor.txt").exists()
+      jsonCfgContent <- JSON.createDatasetCfg(additionalLicenseFilenameAndMimeType, audiences, agreementsXmlExists, messageFromDepositorExists)
       _ <- writeAMD(sdoDir, amdContent.toString())
       _ <- writeFoxml(sdoDir, foxmlContent)
       _ <- writePrsql(sdoDir, PRSQL.create())
+      _ <- writeBagMetadata(sdoDir, agreementsXmlExists, messageFromDepositorExists)
       _ <- writeJsonCfg(sdoDir, jsonCfgContent)
     } yield (emdContent, amdContent) // easy-ingest-flow hands these over to easy-ingest
   }
 
+  private def writeBagMetadata(sdoDir: File, agreementsXmlExists: Boolean, messageFromDepositorExists: Boolean)(implicit s: Settings): Try[Unit] = {
+    if (s.includeBagMetadata) {
+      for {
+        _ <- readFile("metadata/dataset.xml").flatMap(writeDatasetXML(sdoDir, _))
```

Comment:
Perhaps a Boolean argument mandatory for readFile?

See also `Option.when` in scala 2.13 https://stackoverflow.com/questions/19690531/scala-boolean-to-option

Reply:
Perhaps in the future. Keep in mind that we're still on 2.12!